tim1/src/main/java/isamrs/tim1/service/ReservationService.java

```diff
@@ -386,5 +392,56 @@ private void inviteFriendToFlight(String email, FlightReservationDTO flightRes,
 		return new ResponseEntity<MessageDTO>(
 				new MessageDTO("Successfully declined reservation.", ToasterType.SUCCESS.toString()), HttpStatus.OK);
 	}
+	
+	public MessageDTO createQuickFlightReservation(QuickFlightReservationDTO quickDTO) {
+		Flight f = flightRepository.findOneByFlightCode(quickDTO.getFlightCode());
+		if (f == null) {
+			return new MessageDTO("Flight does not exist.", ToasterType.ERROR.toString());
+		}
+		QuickFlightReservation qfr = new QuickFlightReservation();
+		double price = 0.0;
+		String[] idx = quickDTO.getSeat().split("_");
+		int row = Integer.parseInt(idx[0]);
+		int column = Integer.parseInt(idx[1]);
+		if (checkIfSeatIsReserved(f, row, column)) {
+			return new MessageDTO("One of the seats is already reserved.", ToasterType.ERROR.toString());
+		}
+		Seat st = new Seat();
+		st.setRow(row);
+		st.setColumn(column);
+		if (idx[2].equalsIgnoreCase(PlaneSegmentClass.FIRST.toString().substring(0, 1))) {
+			st.setPlaneSegment(new PlaneSegment(PlaneSegmentClass.FIRST));
+			price = f.getFirstClassPrice();
+		} else if (idx[2].equalsIgnoreCase(PlaneSegmentClass.BUSINESS.toString().substring(0, 1))) {
+			st.setPlaneSegment(new PlaneSegment(PlaneSegmentClass.BUSINESS));
+			price = f.getBusinessClassPrice();
+		} else if (idx[2].equalsIgnoreCase(PlaneSegmentClass.ECONOMY.toString().substring(0, 1))) {
+			st.setPlaneSegment(new PlaneSegment(PlaneSegmentClass.ECONOMY));
+			price = f.getEconomyClassPrice();
+		}
+		PassengerSeat ps = new PassengerSeat(new PassengerDTO("", "", "", 0), st);
+		qfr.setFlight(f);
+		qfr.setDone(false);
+		qfr.setDiscount(Integer.parseInt(quickDTO.getDiscount()));
+		qfr.setPrice(price);
+		ps.setReservation(qfr);
+		qfr.getPassengerSeats().add(ps);
+		f.getAirline().getReservations().add(qfr);
+		quickFlightReservationRepository.save(qfr);
+		System.out.println(qfr.getId());
+		return new MessageDTO("Quick flight reservation successfully created", ToasterType.SUCCESS.toString());
+	}
 
+	public ArrayList<QuickFlightReservationDTO> getQuickFlightReservations() {
+		AirlineAdmin airlineAdmin = (AirlineAdmin) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+		Airline a = airlineAdmin.getAirline();
+		ArrayList<QuickFlightReservationDTO> quickRes = new ArrayList<QuickFlightReservationDTO>();
+		Set<FlightReservation> fRes = a.getReservations();
+		for(FlightReservation fr : fRes) {
+			if (fr instanceof QuickFlightReservation) {
```

Comment:
Add this check in _if condition_: fr.getUser()==null

Reply:
Corrected